sql/src/main/java/io/crate/expression/reference/sys/check/node/JvmVersionNodeCheck.java

```diff
@@ -71,7 +71,8 @@ public boolean validate() {
             return new int[] {major, minor, hotfix};
         } catch (NumberFormatException e) {
             if (firstToken.endsWith("ea")) {
-                int major = Integer.parseInt(firstToken.substring(0, firstToken.length() - 2));
+                int offset = firstToken.endsWith("-ea") ? 3 : 2;
```

Comment:
How about trimming any non-numeric characters and return the rest?
Another thought would be to capture this with a regular expression treating  '-' as optional

Not sure if the complexity justifies it - it's just that I found first checking for "es" and then for "-es", 
a bit strange

Reply:
I'll keep it as it is. I guess there would be many ways to parse this. I can see that changing it to trim non-numeric characters would make it more lenient, but then I'd probably also have tests to verify this more lenient behavior which would lead to having to come up with more version variants - which I'm not sure exist. I added tests for the common cases I saw based on our jdk installations.